src/lib/routes/client-api/feature.ts

```diff
@@ -130,39 +119,30 @@ export default class FeatureController extends Controller {
             ],
         });
 
-        if (clientFeatureCaching?.enabled) {
-            this.cache = true;
-            this.cachedFeatures = memoizee(
-                (query) => this.resolveFeaturesAndSegments(query),
+        if (clientFeatureCaching.enabled) {
+            this.featuresAndSegments = memoizee(
+                // eslint-disable-next-line @typescript-eslint/no-unused-vars
+                (query: IFeatureToggleQuery, etag: string) =>
+                    this.resolveFeaturesAndSegments(query),
                 {
                     promise: true,
-                    maxAge: clientFeatureCaching.maxAge,
+                    maxAge: clientFeatureCaching.maxAge
```

Comment:
do we then default "clientFeatureCaching.maxAge" to undefined? I think this logic should be moved to where we setup all the options instead of in-lining it here.

Reply:
The current default maxAge will probably always win?

Reply:
You mean for clarity, because the linter puts the Elvis operator in 3 lines, right? I think that makes sense

Reply:
Oh, I know what you mean!

Reply:
Check out: https://github.com/Unleash/unleash/pull/3665/commits/11e71ade467ddddda4a811c747556f0729f936bc